### PR TITLE
Scan summary for local scanners

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: model/src/main/kotlin/ScanResult.kt [hunk 1]
```diff
@@ -52,31 +50,12 @@ data class ScanResult(
...
-            licenseFindings.mapTo(set) { it.location.path }
-            copyrightFindings.mapTo(set) { it.location.path }
-        }.size
-
-        val summary = summary.copy(
-            fileCount = fileCount,
-            licenseFindings = licenseFindings,
-            copyrightFindings = copyrightFindings
-        )
+        val summary = summary.filterByPath(path)
```
Comment (fviernau) [inline model/src/main/kotlin/ScanResult.kt:58]: commit-msg: our -> out

File: model/src/main/kotlin/utils/FileArchiver.kt [hunk 2]
```diff
@@ -48,7 +48,7 @@ class FileArchiver(
      * A list of globs to match the paths of files that shall be archived. For details about the glob pattern see
      * [java.nio.file.FileSystem.getPathMatcher].
      */
-    patterns: List<String>,
+    patterns: Collection<String>,
```
Comment (fviernau) [inline model/src/main/kotlin/utils/FileArchiver.kt:51]: Have you considered updating the comment? e.g. "A list.."

**Review by fviernau — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/ScanResult.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/ScanResult.kt:58]: The typo is still there...

File: model/src/main/kotlin/utils/FileArchiver.kt (hunk 2 above)
Comment (sschuberth) [reply model/src/main/kotlin/utils/FileArchiver.kt:51]: Done.

File: model/src/main/kotlin/ScanResult.kt (hunk 1 above)
Comment (sschuberth) [reply model/src/main/kotlin/ScanResult.kt:58]: I didn't re-request a review until now as I had only fixed it locally.

File: utils/src/main/kotlin/FileMatcher.kt [hunk 3]
```diff
@@ -30,7 +30,7 @@ class FileMatcher(
      *
      * [1]: https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob
      */
-    val patterns: List<String>,
+    val patterns: Collection<String>,
```
Comment (fviernau) [inline utils/src/main/kotlin/FileMatcher.kt:33]: Sorry, noticed just now that this KDoc probably needs similar change as above.

**Review by fviernau — APPROVED** [review]

**Pull request merged**